wasp/src/main/java/com/orhanobut/wasp/VolleyImageHandler.java

```diff
@@ -0,0 +1,135 @@
+        final String url = waspImage.getUrl();
+        final int defaultImage = waspImage.getDefaultImage();
+
+        int width = imageView.getMeasuredWidth();
+        int height = imageView.getMeasuredHeight();
+
+        boolean wrapWidth = false;
+        boolean wrapHeight = false;
+        if (imageView.getLayoutParams() != null) {
+            wrapWidth = imageView.getLayoutParams().width == ViewGroup.LayoutParams.WRAP_CONTENT;
+            wrapHeight = imageView.getLayoutParams().height == ViewGroup.LayoutParams.WRAP_CONTENT;
+        }
+
+        // if there was an old request in this view, check if it needs to be canceled.
+        if (imageContainer != null && imageContainer.getRequestUrl() != null) {
+            if (imageContainer.getRequestUrl().equals(url)) {
+                // if the request is from the same URL, return.
+                return;
+            } else {
+                // if there is a pre-existing request, cancel it if it's fetching a different URL.
+                imageContainer.cancelRequest();
+                setDefaultImage(defaultImage, imageView);
+            }
+        }
+
+        // Calculate the max image width / height to use while ignoring WRAP_CONTENT dimens.
+        int maxWidth = wrapWidth ? 0 : width;
+        int maxHeight = wrapHeight ? 0 : height;
+
+        imageContainer = imageLoader.get(url, new WaspImageListener(imageView, waspImage), maxWidth, maxHeight);
+    }
+
+    private void setDefaultImage(int placeHolder, ImageView imageView) {
+        if (placeHolder != 0) {
+            imageView.setImageResource(placeHolder);
+            return;
+        }
+        imageView.setImageBitmap(null);
+    }
+
+    private static class WaspImageListener implements ImageLoader.ImageListener {
+
+        final ImageView imageView;
+        final WaspImage waspImage;
+
+        private WaspImageListener(ImageView imageView, WaspImage waspImage) {
+            this.imageView = imageView;
+            this.waspImage = waspImage;
+        }
+
+        @Override
+        public void onResponse(final ImageLoader.ImageContainer response, boolean isImmediate) {
+            // If this was an immediate response that was delivered inside of a layout
+            // pass do not set the image immediately as it will trigger a requestLayout
+            // inside of a layout. Instead, defer setting the image by posting back to
+            // the main thread.
+            if (isImmediate) {
+                imageView.post(new Runnable() {
+                    @Override
+                    public void run() {
+                        onResponse(response, false);
+                    }
+                });
+                return;
+            }
+
+            if (response.getBitmap() != null) {
+                imageView.setImageBitmap(response.getBitmap());
+            } else if (waspImage.getDefaultImage() != 0) {
+                imageView.setImageResource(waspImage.getDefaultImage());
+            }
```

Comment:
I would write like this:
<pre>
Bitmap bitmap = response.getBitmap(); 
if (bitmap != null) {
    imageView.setImageBitmap(bitmap);
    return;
}
 
int defaultImage = waspImage.getDefaultImage();
if (defaultImage != 0) {
    imageView.setImageResource(defaultImage); 
}
</pre>